rpc/jsonrpc/server/http_json_handler.go

```diff
@@ -57,6 +57,9 @@ func makeJSONRPCHandler(funcMap map[string]*RPCFunc, logger log.Logger) http.Han
 			requests = []types.RPCRequest{request}
 		}
 
+		// Set the default response cache to true unless we see any error and any rpc request doesn't
+		// allow to be cached
+		var c = true
```

Comment:
There is one other condition that we have to be wary about. For a lot of RPC requests i.e validators, consensus params, blocks and commits, if a client requests this data with a height of 0 the node responds by sending that data type at the latest height. This means that the same request of height 0 to these RPC endpoints should also over time send different responses and thus shouldn't cache the response

Reply:
That is correct!

Reply:
Thanks for the feedback, I will check the condition and revise the logic.

Reply:
Awesome. Once this is done I think the PR should be ready to merge

Reply:
Added default height checking for these types of RPC requests.